extensions/binding/org.eclipse.smarthome.binding.openweathermap/ESH-INF/config/config.xml

```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<config-description:config-descriptions
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:config-description="http://eclipse.org/smarthome/schemas/config-description/v1.0.0"
+	xsi:schemaLocation="http://eclipse.org/smarthome/schemas/config-description/v1.0.0 http://eclipse.org/smarthome/schemas/config-description-1.0.0.xsd">
+
+	<!-- Config for OpenWeatherMap Binding -->
+	<config-description uri="bridge-type:openweathermap:weather-api">
+		<parameter name="apikey" type="text" required="true">
+			<context>password</context>
+			<label>API Key</label>
+			<description>API key to access the OpenWeatherMap API: https://home.openweathermap.org/users/sign_up.</description>
+		</parameter>
+		<parameter name="refreshInterval" type="integer" min="10" unit="min">
+			<label>Refresh Interval</label>
+			<description>Specifies the refresh interval (in minutes).</description>
+			<default>60</default>
+		</parameter>
+		<parameter name="language" type="text">
+			<label>Language</label>
+			<description>Language to be used by the OpenWeatherMap API.</description>
+			<options>
+				<option value="ar">Arabic</option>
+				<option value="bg">Bulgarian</option>
+				<option value="ca">Catalan</option>
+				<option value="zh_cn">Chinese - Simplified</option>
+				<option value="zh_tw">Chinese - Traditional</option>
+				<option value="hr">Croatian</option>
+				<option value="cz">Czech</option>
+				<option value="nl">Dutch</option>
+				<option value="en">English</option>
+				<option value="fi">Finnish</option>
+				<option value="fr">French</option>
+				<option value="gl">Galician</option>
+				<option value="de">German</option>
+				<option value="el">Greek</option>
+				<option value="hu">Hungarian</option>
+				<option value="it">Italian</option>
+				<option value="ja">Japanese</option>
+				<option value="kr">Korean</option>
+				<option value="la">Latvian</option>
+				<option value="lt">Lithuanian</option>
+				<option value="mk">Macedonian</option>
+				<option value="fa">Persian (Farsi)</option>
+				<option value="pl">Polish</option>
+				<option value="pt">Portuguese</option>
+				<option value="ro">Romanian</option>
+				<option value="ru">Russian</option>
+				<option value="sk">Slovak</option>
+				<option value="sl">Slovenian</option>
+				<option value="es">Spanish</option>
+				<option value="sw">Swedish</option>
+				<option value="tr">Turkish</option>
+				<option value="ua">Ukrainian</option>
+				<option value="vi">Vietnamese</option>
+			</options>
+		</parameter>
+		<parameter name="hourlyForecast" type="boolean">
+			<label>3 Hour Forecast</label>
+			<description>Enable / Disable discovery of the 3 hour forecast thing.</description>
```

Comment:
I second your opinion on the description. I will think about it.

The way how I modeled bridges and things has not been discussed in detail with others yet. The idea behind my model is mainly based in the different forecast options provided by OWM and the underlying accounting / price model (The hourly forecast is free for everyone and the daily forecast restricted to payed accounts).
Additionally I am planning to do some extensions once this will have been merged. OWM provides UV Index, Air pollution and weather alerts too. I want to add them as new things too because I do not think that everyone needs them or is allowed to access them. Why overload one thing with channels / data which are not necessary.

Reply:
> Why overload one thing with channels / data which are not necessary?

Because Things are not a means to express a functional structure, but [are the entities that can physically be added to a system and which can potentially provide many functionalities in one](https://www.eclipse.org/smarthome/documentation/concepts/things.html).

Certainly you should only add the channels to the Thing that really ARE available. So if it is a free account, the daily forecast channels would not be added. Btw, can this be determined automatically?

Reply:
> Btw, can this be determined automatically?

OWM does not expose any information about the account. I can try to ping the daily forecast API and evaluate the result to determine the details of the account. But we cannot say it for 100% sure.

Reply:
My current approach tries to request daily forecast data from the OWM API. If the request fails, all channel groups are removed from the thing and further request are omitted.